projects/plugins/jetpack/_inc/client/search/dashboard/mocked-search.jsx

```diff
@@ -0,0 +1,43 @@
+/**
+ * External dependencies
+ */
+import React, { Fragment } from 'react';
+import { connect } from 'react-redux';
+
+/**
+ * Internal dependencies
+ */
+import MockedInstantSearch from './mocked-instant-search';
+import MockedLegacySearch from './mocked-legacy-search';
+import { getPlanClass } from 'lib/plans/constants';
+
+/**
+ * State dependencies
+ */
+import { getSitePlan, hasActiveSearchPurchase as selectHasActiveSearchPurchase } from 'state/site';
+
+/**
+ * Mocked Search component, which shows mocked Instant Search or legacy Search interface.
+ *
+ * @param {object} props - Component properties.
+ * @returns {React.Component} Mocked Search interface component.
+ */
+function MockedSearch( props ) {
+	const { hasActiveSearchPurchase, isBusinessPlan } = props;
+	const shouldShowMockedInstantSearch = ! isBusinessPlan || hasActiveSearchPurchase;
```

Comment:
I'm not sure I understand this conditional. Couldn't we rely purely on `hasActiveSearchPurchase` for this check?

If a site has both the business plan and a search subscription, wouldn't we want to show the Instant Search mock?

Reply:
It is a bit confusing yes, I'll add some comments there. Basically we want to show the 'legacy' mock only to the users who has business plan but no search subscription, meaning we want to show legacy mock as little as possible. For others, we show the instant search mock, our current gen product. The logic might be easier to understand if written in this way `!(isBusinessPlan && !hasActiveSearchPurchase)`

Reply:
changed in 1dd604081d

Reply:
So we have the following possibilities:

1. Has the business plan and the search subscription.
2. Has the business plan, but no search plan.
3. Has the search subscription, but no business plan.
4. Has neither the business plan nor the search subscription.

For 1 & 3, we'd want to show the instant search mock; for 2, we'd want to show the old mock; we can ignore 4 because they won't be able to see this page.

Wouldn't setting `const shouldShowMockedInstantSearch = hasActiveSearchPurchase;` adequately cover our use-case here (1, 2, and 3)?

> Basically we want to show the 'legacy' mock only to the users who has business plan but no search subscription, meaning we want to show legacy mock as little as possible. For others, we show the instant search mock, our current gen product. The logic might be easier to understand if written in this way `!(isBusinessPlan && !hasActiveSearchPurchase)` @kangzj 

Using `const shouldShowLegacyMock = isBusinessPlan && !hasActiveSearchPurchase` would make this value easier to comprehend.

Reply:
>Wouldn't setting const shouldShowMockedInstantSearch = hasActiveSearchPurchase; adequately cover our use-case here (1, 2, and 3)?

yep

> Using const shouldShowLegacyMock = isBusinessPlan && !hasActiveSearchPurchase would make this value easier to reason about.

That's right mate, just done that :-)